aptos-move/aptos-vm-types/src/abstract_write_op.rs

```diff
@@ -85,28 +85,29 @@ impl AbstractResourceWriteOp {
             | WriteResourceGroup(GroupWrite {
                 metadata_op: write_op,
                 ..
-            })
+            }) => write_op.get_metadata_mut(),
+            InPlaceDelayedFieldChange(InPlaceDelayedFieldChangeOp { metadata, .. })
```

Comment:
Does this metadata state that the resource has a delayed field inside it? What's the benefit of storing this in the metadata of the resource?

Reply:
Storing metadata isn't related to this change but it is for e.g. storage fee accounting purposes - needed when the transaction is finalized and can't be postponed until the whole group write is prepared by the block executor. We previously had a WriteOp and populated the bytes later, but only needed metadata. Now it's only metadata (and gets combined with finalized bytes/data to create WriteOp later)

Reply:
> it is for e.g. storage fee accounting purposes
So, amount of money charged for a resource depends on whether the resource has a delayed field inside it? In that case, we should be storing the number of delayed fields inside the resource right?

> needed when the transaction is finalized and can't be postponed until the whole group write is prepared by the block executor. 
I didn't get this. I thought if a transaction if finalized, that means the block executor finished running the transaction and committed the transaction output. We compute the group write set after the transaction is finalized?

Reply:
we compute the actual bytes that go to storage in the block executor now, at this point transaction is finished.
When transaction is finishing it needs access to the metadata. that's the essence of it, how storage deposit etc works should be fully orthogonal.